### PR TITLE
feat: add loop of the animation badge

### PR DESCRIPTION
**Pull request by sawel24**
## Connection with issue(s)

Resolve issue https://github.com/yako-dev/flutter_badges/issues/66

Adds a loop of the animation badge

https://user-images.githubusercontent.com/82770717/199070446-b7a3e74f-76c4-49a7-a9af-0abc47129c00.mov

### CODE REVIEW

**Comment by yadaniyil** [comment]
@sawel24 no one will use this type of loop animation. Can you show how fade in fade out looks like?

**Comment by sawel24** [comment]
> @sawel24 no one will use this type of loop animation. Can you show how fade in fade out looks like?
These are examples of looped fade animation and scale animation

https://user-images.githubusercontent.com/82770717/199462210-4d02a43e-31b0-490c-83f7-033cf81ec69a.mov


https://user-images.githubusercontent.com/82770717/199462283-8c13c3e4-2a00-4e50-a54f-c0ba61395651.mov

File: lib/src/badge.dart [hunk 1]
```diff
@@ -139,6 +140,17 @@ class Badge extends StatefulWidget {
   /// If true, the badge will be displayed, if false, it won't.
   final bool showBadge;
 
+  ///Controls loop of the animation
```
Comment (yadaniyil) [inline lib/src/badge.dart:143]: there have to be a blank space

File: lib/src/badge.dart [hunk 2]
```diff
@@ -139,6 +140,17 @@ class Badge extends StatefulWidget {
...
+  ///
+  /// The default value is false
+  ///
+  /// /// See also:
```
Comment (yadaniyil) [inline lib/src/badge.dart:147]: why 6 slashes are used?

**Comment by sawel24** [comment]
New versions of the animation

https://user-images.githubusercontent.com/82770717/199479714-b3242f9d-d42c-463a-b2ae-9792ccbf37a8.mov

https://user-images.githubusercontent.com/82770717/199479624-87bb8f41-41f9-4e6c-a48c-2a43d6f909a7.mov

**Comment by yadaniyil** [comment]
@sawel24 Good job! Can we try to add curves to our animations? They will look more real

**Comment by sawel24** [comment]
> @sawel24 Good job! Can we try to add curves to our animations? They will look more real

I added a variable to control our animation. These are one of the examples, animation with adding a curve 

https://user-images.githubusercontent.com/82770717/199551570-a93b25e1-144b-458e-a747-4c197f4e300c.mov

**Comment by yadaniyil** [comment]
@sawel24 In this PR you are breaking the current animation behavior. You make Curve.easeIn the default curve, but you removed Curves.elasticOut for slide animation type. 
Also, think about creating a separate constructor for looped animation badge, e.g. Badge.looped(child:...)

**Review by yadaniyil — CHANGES_REQUESTED** [review]
Left a comment

**Comment by sawel24** [comment]
> @sawel24 In this PR you are breaking the current animation behavior. You make Curve.easeIn the default curve, but you removed Curves.elasticOut for slide animation type. Also, think about creating a separate constructor for looped animation badge, e.g. Badge.looped(child:...)

I fixed the broken animation for the slide and created a separate constructor for the looped Badge. Recheck the code, is this the right solution?

**Review by yadaniyil — APPROVED** [review]

**Pull request merged**